PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/core/FunctionUtils.java

```diff
@@ -0,0 +1,375 @@
+            FileUtils.write(result, DEFAULT_HOST_JSON, Charset.defaultCharset());
+            return result.toPath();
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    public static void prepareStagingFolderForDeploy(Path stagingFolder, Path hostJson, Module module, PsiMethod[] methods)
+            throws AzureExecutionException, IOException {
+        prepareStagingFolder(stagingFolder, hostJson, null, module, methods);
+    }
+
+    private static void copyFilesWithDefaultContent(Path sourcePath, File dest, String defaultContent) throws IOException {
+        final File src = sourcePath == null ? null : sourcePath.toFile();
+        if (src != null && src.exists()) {
+            FileUtils.copyFile(src, dest);
+        } else {
+            FileUtils.write(src, defaultContent, Charset.defaultCharset());
+        }
+    }
+
+    public static void prepareStagingFolder(Path stagingFolder, Path hostJson, Path localSettingJson, Module module,
+                                            PsiMethod[] methods) throws AzureExecutionException, IOException {
+        final Map<String, FunctionConfiguration> configMap = generateConfigurations(methods);
+        final Path jarFile = JarUtils.buildJarFileToStagingPath(stagingFolder.toString(), module);
+        final String scriptFilePath = "../" + jarFile.getFileName().toString();
+        configMap.values().forEach(config -> config.setScriptFile(scriptFilePath));
+        for (final Map.Entry<String, FunctionConfiguration> config : configMap.entrySet()) {
+            if (StringUtils.isNotBlank(config.getKey())) {
+                final File functionJsonFile = Paths.get(stagingFolder.toString(), config.getKey(), FUNCTION_JSON)
+                        .toFile();
+                FunctionJsonWriter.writeFunctionJsonFile(functionJsonFile, config.getValue());
+            }
+        }
+
+        copyFilesWithDefaultContent(hostJson, new File(stagingFolder.toFile(), "host.json"), DEFAULT_HOST_JSON);
+        copyFilesWithDefaultContent(localSettingJson, new File(stagingFolder.toFile(), "local.settings.json"), DEFAULT_LOCAL_SETTINGS_JSON);
+
+        final List<File> jarFiles = new ArrayList<>();
+        OrderEnumerator.orderEntries(module).productionOnly().forEachLibrary(lib -> {
+            if (StringUtils.contains(lib.getName(), FUNCTION_JAVA_LIBRARY_ARTIFACT_ID)) {
+                return true;
+            }
+
+            if (lib != null) {
+                for (final VirtualFile virtualFile : lib.getFiles(OrderRootType.CLASSES)) {
+                    final File file = new File(stripExtraCharacters(virtualFile.getPath()));
+                    if (file.exists()) {
+                        jarFiles.add(file);
+                    }
+                }
+            }
+            return true;
+        });
+
+        for (final File file : jarFiles) {
+            FileUtils.copyFileToDirectory(file, new File(stagingFolder.toFile(), "lib"));
```

Comment:
extract  new File(stagingFolder.toFile(), "lib") to a var

Reply:
Done